src/firewall/core/fw_nm.py

```diff
@@ -49,6 +50,22 @@ def nm_get_client():
     global _nm_client
     if not _nm_client:
         _nm_client = NM.Client.new(None)
+
+        def _release():
+            """
+            Release the client to avoid excess memory usage when libnm pushes
+            irrelevant (to firewalld) updates.
+            """
+            global _nm_client
+            global _nm_client_timeout
+            _nm_client = None
+            _nm_client_timeout = None
+
+        global _nm_client_timeout
+        if _nm_client_timeout:
+            GLib.source_remove(_nm_client_timeout)
```

Comment:
I think `if _nm_client_timeout` cannot be True here.

The only question is, whether we want to restart the timer with full 5 seconds, each time when `nm_get_client()` is called. Probably not, otherwise can just just move these few lines after the `if not _nm_client` block.

Reply:
rewrote, please review again.